plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNResource.java

```diff
@@ -144,9 +147,21 @@ public String getNodeDescription() {
         return handler == null || resource == null ? null : handler.getResourceDescription(resource);
     }
 
+    @NotNull
     @Override
-    public DBPImage getNodeIcon()
-    {
+    public DBPImage getNodeIcon() {
+        DBPImage iconImage = this.getResourceNodeIcon();
+
+        DBPProject project = getOwnerProject();
+        if (project != null && !project.hasRealmPermission(RMConstants.PERMISSION_PROJECT_RESOURCE_EDIT)) {
+            iconImage = new DBIconComposite(iconImage, false, null, null, null, DBIcon.OVER_LOCK);
+        }
+
+        return iconImage;
```

Comment:
```suggestion
        DBPProject project = getOwnerProject();
        if (project != null && !project.hasRealmPermission(RMConstants.PERMISSION_PROJECT_RESOURCE_EDIT)) {
            return new DBIconComposite(iconImage, false, null, null, null, DBIcon.OVER_LOCK);
        }

        return getResourceNodeIcon();
```

Reply:
I don't like multiple exit points in short methods.

Reply:
It's better than not using result of method call.

Reply:
I looks closer and I can confirm that I can't do what you suggest. I need the result of `this.getResourceNodeIcon()` method to create composite image (the first argument of DBIconComposite ctor).

Reply:
Oh, now I see. My apologies for giving inappropriate suggestion.